packages/extension-api/src/extension-api.d.ts

```diff
@@ -3715,6 +3715,16 @@ declare module '@podman-desktop/api' {
      * The system clipboard.
      */
     export const clipboard: Clipboard;
+
+    /**
+     * Indicates if the user is using WSL to run podman machines
+     */
+    export function isWSL(): Promise<boolean>;
+
+    /**
+     * Indicates if the user is using HyperV to run podman machines
+     */
+    export function isHyperV(): Promise<boolean>;
```

Comment:
I am wondering as WSL and HyperV are podman providers, we should not have a more generic method like

```
    /**
     * Indicates if the user is using HyperV to run podman machines
     * @return the provider (E.g. qemu, applehv, hyperv, wsl)
     */
     export function getPodmanMachineProvider(machineId?: string): Promise<string>
```

Reply:
I thought the same at first but then i thought that starting from podman 5 we only support applehv on mac so isMac is good enough for macOs and this generic method would only return windows stuff.

Reply:
Even tho on MacOS we have in v5 a single value, in v4 it is not the case, and in the future, we might have others (we never know).

For example, Podman still uses a switch cases, https://github.com/containers/podman/blob/9697a5f08be544631c3b1b9889b906237e133c35/pkg/machine/provider/platform.go#L31-L35 they might add another later. For the GPU they were talking about creating a custom provider.